src/supernode/src/test/java/com/alibaba/dragonfly/supernode/repository/ProgressRepositoryTest.java

```diff
@@ -0,0 +1,157 @@
+package com.alibaba.dragonfly.supernode.repository;
+
+import static org.mockito.AdditionalMatchers.or;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Matchers.isNull;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+import com.alibaba.dragonfly.supernode.repository.PieceState;
+import com.alibaba.dragonfly.supernode.repository.ProgressRepository;
+import com.diffblue.deeptestutils.Reflector;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.BitSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@PowerMockIgnore({"javax.management.*"})
+@RunWith(PowerMockRunner.class)
+public class ProgressRepositoryTest {
```

Comment:
All tests in this class are Exceptional 👍

Reply:
👍